Sources/Apollo/GraphQLQueryWatcher.swift

```diff
@@ -66,7 +69,21 @@ public final class GraphQLQueryWatcher<Query: GraphQLQuery>: Cancellable, Apollo
     guard let dependentKeys = dependentKeys else { return }
 
     if !dependentKeys.isDisjoint(with: changedKeys) {
-      fetch(cachePolicy: .returnCacheDataElseFetch)
+      // First, attempt to reload the query from the cache directly, in order not to interrupt any in-flight server-side fetch.
+      store.load(query: query) { [weak self] result in
+        guard let self = self else { return }
+
+        switch result {
+        case .success(let graphQLresult):
+          self.queue.async {
+            self.dependentKeys = graphQLresult.dependentKeys
```

Comment:
I don't think this should be necessary - you have to call `fetch` on a watcher when you fire it up, and at that point `dependentKeys` would be set by the value returned from the `fetch` method. This would only get called if you initialized the watcher and failed to call the initial `fetch` method, which is not something we want to encourage.

Reply:
Actually this is necessary. If I create a watcher on query `A` and perform another query `B` if I want to update manually `A` in the store with results from `B`  subsequent updates to the updated elements from `B` wouldn't be propagated to `A`'s listeners.

Use case pagination using apollo store.

Watch primary query, perform fetch more and update primary with nodes, page info and has next, then perform mutation in a node that came from fetch more query, primary query's listeners won't retrieve the updated result even though they should.